participant/tests/test_api.py

```diff
@@ -1,23 +1,92 @@
+from participant.serializer import ParticipantSerializer
 from phishtray.test.base import PhishtrayAPIBaseTest
-from ..factories import ParticipantFactory
+from ..factories import ParticipantFactory, ParticipantActionFactory, ProfileEntryFactory
 from exercise.factories import DemographicsInfoFactory
 
 
+class ParticipantAPITests(PhishtrayAPIBaseTest):
+    
+    def setUp(self):
+        super(ParticipantAPITests, self).setUp()
+    
+    def test_participant_list_block_public(self):
+        """
+        Non admin users should not be able to retrieve participant list.
+        """
+        url = reverse('api:participant-list')
+        response = self.client.get(url)
+
+        self.assertEqual(status.HTTP_403_FORBIDDEN, response.status_code)
+        
+    def test_participant_list_allow_admin(self):
+        """
+        Admin users should be able to retrieve thread list.
+        """
+        url = reverse('api:participant-list')
+        participant_count = randint(1, 50)
+        ParticipantFactory.create_batch(participant_count)
+
+        response = self.admin_client.get(url)
+        serialized = ParticipantSerializer(Participant.objects.all(), many=True)
+
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertEqual(participant_count, len(response.data))
+        self.assertEqual(serialized.data, underscoreize(response.data))
+        
+    def test_participant_details(self):
+        """
+        Participant details are public.
+        """
+        participants_count = randint(1, 10)
+        ParticipantFactory.create_batch(participants_count)
+        participant = ParticipantFactory()
+        
+        # Actions
+        actions_count = randint(1, 10)
+        ParticipantActionFactory.create_batch(actions_count, participant=participant)
+        
+        # Profile Entries
+        profile_entries_count = randint(1, 10)
+        profile_entries = ProfileEntryFactory.create_batch(profile_entries_count)
+        participant.profile.add(*profile_entries)
+        participant.save()
+
+        url = reverse('api:participant-detail', args=[participant.id])
+        response = self.admin_client.get(url)
+        serialized = ParticipantSerializer(Participant.objects.get(pk=participant.id))
+
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertEqual(serialized.data, underscoreize(response.data))
+        self.assertEqual(actions_count, len(response.data.get('actions')))
+        self.assertEqual(profile_entries_count, len(response.data.get('profile')))
+    
+    def test_get_participant_details_404(self):
+        ParticipantFactory.create_batch(5)
+        fake_id = 'fakeID'
+        url = reverse('api:participant-detail', args=[fake_id])
+
+        response = self.client.get(url)
+
+        self.assertEqual(status.HTTP_404_NOT_FOUND, response.status_code)
+        self.assertEqual('Not found.', response.data.get('detail'))
```

Comment:
is this assertion necessary/different from the above status code check

Reply:
There's no burning need to include it, but if you were to display the Error message at the FE at any time it's good to have this check in.

I'm happy to take the one out on line 79 if you have strong feelings about not to leave it in there.